### ISSUE TITLE
`GitClient.get_refs()` doesn't return peeled refs anymore (git protocol v2)

### ISSUE THREAD

**Issue by abderrahim** [jelmer/dulwich#1410]
The following code snipped prints 198 when ran against dulwich 0.22.1 and 0 when ran against latest 0.22.4.

```
from dulwich.client import get_transport_and_path
from dulwich.refs import ANNOTATED_TAG_SUFFIX

client, path = get_transport_and_path("https://github.com/jelmer/dulwich.git")
refs = client.get_refs(path)

print(len([k for k in refs.keys() if k.endswith(ANNOTATED_TAG_SUFFIX)]))
```

The issue seems to be git protocol version 2. In the newer protocol, [`ls-refs`](https://git-scm.com/docs/protocol-v2#_ls_refs) command needs a special argument `peel` to return peeled tags.

While I think ultimately it would be a good idea to add more flexibility to getting refs afforded by protocol version 2 (either by adding optional arguments to the current method, or having a new method), the default behaviour should closely mirror what you get with protocol version 1.

**Comment by nanonyme** [comment]
Right, this is fairly large functional change and if desired, should probably happen in 0.23.0, not 0.22.4. @jelmer any thoughts?

**Comment by jelmer** [comment]
cc @stspdotname, as this relates to v2.

I'd argue that actually this is regression and that we should fix this in 0.22. I don't think it's necessarily a big change.

**Comment by stspdotname** [comment]
Agreed. See the PR #1413
